source/connect/tls.txt

```diff
@@ -0,0 +1,245 @@
+          used to validate certs passed from the other end of the connection.
+          By default, the driver will use the default system root certificate 
+          store as the trust anchor, if the option is not designated. 
+
+        These certificate options can be specified as different types. For example, the 
+        ``ssl_cert`` option has the alternative ``ssl_cert_object`` and ``ssl_cert_string`` 
+        options, to specify the certificate file as either a certificate object or a string.
+        For more information on certificate options, see the Options Hash section of 
+        the API Documentation for `Mongo::Client <{+api-root+}/Mongo/Client.html>`__ .  
+
+        In the following example, the TLS certificate and corresponding private key 
+        are provided in separate files:
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-settings
+            :end-before: end-enable-tls-settings
+        
+        The TLS certificate and private key can be contained in the same file, but 
+        both the certificate and private key options must still be specified. In the 
+        following example, the TLS certificate and the private key are both defined 
+        in the same ``client.pem`` file:
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-settings-same-file
+            :end-before: end-enable-tls-settings-same-file
+
+    .. tab:: Connection String
+        :tabid: uri
+
+        To configure certificates, you must specify the following URI options:
+        
+        - ``tlsCertificateKeyFile``: The certificate and key file used to identify
+          identify the connection against MongoDB. Both the certificate and key
+          must be stored in the same file provided to this option.
+        - ``tlsCAFile``: The file containing the concatenated certificate authority (CA)
+          certificates used to validate certificates passed from the other end of the 
+          connection. The driver will use the default system root certificate 
+          store as the trust anchor, if the option is not designated. 
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-uri
+            :end-before: end-enable-tls-uri
+        
+        The file containing the certificate and key usually has a``.crt`` or ``.pem`` 
+        extension.
+
+        URI option values must be properly URI escaped. This applies, for example, 
+        to slashes (/) in the paths, which are encoded as ``%2f``.
+
+Modify the SSL Context
+----------------------
+
+If your TLS configuration requires customization, you can set TLS context hooks
+by adding a native Ruby class ``Proc`` object to the ``Mongo.tls_context_hooks`` 
```

Comment:
q: is 'Ruby class Proc object' the right description? i would have guessed 'native Ruby Proc object' or similar

Reply:
I think native Ruby Proc object' works, thanks!